### DISCUSSION TITLE
[Feature] Allow album assets to be sorted per-album

### DISCUSSION THREAD

**Discussion by nicokaiser** [immich-app/immich#4800] [Feature Request]
### The feature

Currently, all albums display their photos sorted by Newest first and as a timeline (grouped by date bucket, with years on the right).

In some cases this does not really make sense. It would be great if the sort order could be set on a per-album basis, to allow building albums that "tell a story" (must be in ascending order), e.g.

- server: Add a sort order attribute to the album model ("Oldest first", "Newest first" (default), maybe also "Recently added" and "Title")
- server: Add a global configuration whether to use the timeline view for albums (see below)

Now on the album page:

- Order album assets by the album's sort order
- If the global "timeline in albums" setting is set, and sort order is by a date, display like before (use timeline and date buckets)
- Otherwise, just display all assets without groups and with no timeline scroll bar on the right.

Manual sort order (drag & dropping assets) would require more elaborate changes in the data model and this might be a separate feature. But I think being able to choose ascending or descending order by date would already make the albums feature much more useful already.

Might relate to (or even duplicate?) #4672, #1689, #2662

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by nicokaiser** [comment]
Sorry if this is a duplicate, I did not really find a good matching discussion which addresses timeline view and sorting assets in albums though.

**Comment by tedstriker** [comment]
I was about to open a feature request with the same topic.

Sorting within albums from oldest to newest should be the default imho.

While having the newest images on top, when looking at the entire library - the timeline - is perfectly fine, it feels counter intuitive to sort images within albums from newest to oldest image.
The images in albums behave differently. They usually tell a story. Maybe from a vacation, a celebration, ... and I'd really like to have these stories being told from the beginning on top till the end, at the bottom.
Currently I have to scroll down and do a little mental gymnastics, because it looks like it's the events happen in reverse order.

I would already enjoy the lite-version of your proposal @nicokaiser
Having a configurable, global default for all albums would be an improvement. Even changing the current default would feel more intuitive.

Just my 2 cents

**Comment by bo0tzz** [comment]
Duplicate of #1689
